### DISCUSSION TITLE
How does the cmp integration work?

### DISCUSSION THREAD

**Discussion by mrjones2014** [olimorris/codecompanion.nvim#25] [Q&A] [answered]
I see a `cmp_codecompanion` module in the source tree but I can't find any docs on it

**Comment by olimorris** [comment, marked as answer]
Yeah it's not too fleshed out yet. You'll see if you try to change a model in the settings it will offer completion based on the adapter's schema:

![2024-03-18 13_42_32 - WezTerm@2x](https://github.com/olimorris/codecompanion.nvim/assets/9512444/2e8fab54-0a12-420d-af6a-0937ef53d2db)

**Reply by mrjones2014** [reply]
Ah okay I thought it would like produce completions via AI like Copilot or something.

**Reply by olimorris** [reply]
Oh I've given that some thought...

With Ollama, it makes a lot of sense...with Anthropic or OpenAI, it's super expensive as I imagine it's just the constant sending of tokens to the LLM.

I am keen to work on a feature where you could send the whole code base over to an LLM and ask for a specific feature or bug fix. For an open source plugin that could be really cool

**Reply by mrjones2014** [reply]
> I am keen to work on a feature where you could send the whole code base over to an LLM and ask for a specific feature or bug fix.

This sounds like what [cody](https://github.com/sourcegraph/cody) is. We have it at work, I believe it runs in our own self hosted infrastructure so that it can use our proprietary code as context.

**Reply by olimorris** [reply]
Yes I'd heard of Cody but never tried it out. Copilot Chat has a similar feature now I believe.

Would love to unpick how Cody goes about doing it.
